VideoWeb/VideoWeb/ClientApp/src/app/home/home.component.spec.ts

```diff
@@ -17,14 +18,19 @@ describe('HomeComponent', () => {
     let routerSpy: jasmine.SpyObj<Router>;
     let eventServiceSpy: jasmine.SpyObj<PublicEventsService>;
     let oidcClientNotificationSpy: jasmine.SpyObj<OidcClientNotification<any>>;
+    let routeSpy: jasmine.SpyObj<ActivatedRoute>;
 
     beforeAll(() => {
+        const snapshotMock = new ActivatedRouteSnapshotMock();
```

Comment:
Instead of adding a new class here could you try something like:

```suggestion
        const snapshotSpy = jasmine.createSpyObj('ActivatedRouteSnapshot', [], { queryParamMap: convertToParamMap({}) });
```